pisces/data_sets.py

```diff
@@ -503,7 +503,183 @@ def __init__(self,
+        - 5 => 3 (REM)
+    If N4 is False:
+        - 1, 2 => 1 (light sleep)
+        - 3 => 2 (deep sleep)
+        - 5 => 3 (REM)
+    * retain all 0 (wake) and -1 (mask) classes
+    """
+    return vec_to_WLDM(psg[:, 1].to_numpy(), N4)
+
+# %% ../nbs/01_data_sets.ipynb 23
+class ModelOutputType(Enum):
+    SLEEP_WAKE = auto()
+    WAKE_LIGHT_DEEP_REM = auto()
+
+class PSGType(Enum):
+    NO_N4 = auto()
+    HAS_N4 = auto()
+
+class ModelInput:
+    def __init__(self,
+                 input_features: List[str] | str,
+                 input_sampling_hz: int | float, # Sampling rate of the input data (1/s)
+                 ):
+        # input_features
+        if isinstance(input_features, str):
+            input_features = [input_features]
+        self.input_features = input_features
+        # input_sampling_hz
+        if not isinstance(input_sampling_hz, (int, float)):
+            raise ValueError("input_sampling_hz must be an int or a float")
+        else:
+            if input_sampling_hz <= 0:
+                raise ValueError("input_sampling_hz must be greater than 0")
+        self.input_sampling_hz = float(input_sampling_hz)
+
+class ModelInput1D(ModelInput):
+    def __init__(self,
+                 input_features: List[str] | str,
+                 input_sampling_hz: int | float, # Sampling rate of the input data (1/s)
+                 input_window_time: int | float, # Window size (in seconds) for the input data. Window will be centered around the time point for which the model is making a prediction
+                 ):
+        super().__init__(input_features, input_sampling_hz)
+        # input_window_time
+        if not isinstance(input_window_time, (int, float)):
+            raise ValueError("input_window_time must be an int or a float")
+        else:
+            if input_window_time <= 0:
+                raise ValueError("input_window_time must be greater than 0")
+
+        self.input_window_time = float(input_window_time)
+        # Number of samples for the input window of a single feature
+        self.input_window_samples = int(self.input_window_time * self.input_sampling_hz)
+        ## force it to be odd to have perfectly centered window
+        if self.input_window_samples % 2 == 0:
+            self.input_window_samples += 1
+        # Dimension of the input data for the model
+        self.model_input_dimension = int(len(input_features) * self. input_window_samples)
+
+class ModelInputSpectrogram(ModelInput):
+    def __init__(self,
+                 input_features: List[str] | str,
+                 input_sampling_hz: int | float, # Sampling rate of the input data (1/s)
+                 spectrogram_preprocessing_config: Dict=MO_PREPROCESSING_CONFIG, # Steps in the preprocessing pipeline for getting a spectrogram from acceleration
+                 ):
+        super().__init__(input_features, input_sampling_hz)
+        self.input_sampling_hz = float(input_sampling_hz)
```

Comment:
this is redundant, it should happen from `super().__init__` no? and we can cut the cast to `float` since it is in `super()`'s init as well.